drivers/i2c/i2c_nrfx_twi.c

```diff
@@ -145,19 +149,16 @@ static int init_twi(struct device *dev, const nrfx_twi_config_t *config)
 		.sync = _K_SEM_INITIALIZER(twi_##idx##_data.sync, 0, 1)	\
 	};								\
 	static const struct i2c_nrfx_twi_config twi_##idx##_config = {	\
-		.twi = NRFX_TWI_INSTANCE(idx)				\
+		.twi = NRFX_TWI_INSTANCE(periph_num)			\
 	};								\
-	DEVICE_AND_API_INIT(twi_##idx, CONFIG_I2C_##idx##_NAME,		\
+	DEVICE_AND_API_INIT(twi_##idx, NORDIC_NRF5_I2C_##idx##_LABEL,	\
 			    twi_##idx##_init, &twi_##idx##_data,	\
 			    &twi_##idx##_config, POST_KERNEL,		\
 			    CONFIG_I2C_INIT_PRIORITY,			\
 			    &i2c_nrfx_twi_driver_api)
 
-#ifdef CONFIG_I2C_0_NRF_TWI
-I2C_NRFX_TWI_DEVICE(0);
-#endif
-
-#ifdef CONFIG_I2C_1_NRF_TWI
-I2C_NRFX_TWI_DEVICE(1);
-#endif
-
+#define PERIPH_NUM(i) NORDIC_NRF5_I2C_##i##_PERIPHERAL_NUMBER
+#define DRIVER_INST(i, _)						\
+	_EVAL(NRFX_CONCAT_3(CONFIG_I2C_, PERIPH_NUM(i), _NRF_TWI),	\
+	      (I2C_NRFX_TWI_DEVICE(i, PERIPH_NUM(i));), ())
+UTIL_EVAL(UTIL_REPEAT(NORDIC_NRF5_I2C_COUNT, DRIVER_INST, /*not used */))
```

Comment:
Where is NORDIC_NRF5_I2C_COUNT coming from?

Reply:
This define has count of all enabled devices having the same compatible string calculated from dts.